### PR TITLE
fix crash by clicking on candidates when librime-predict enabled

### PR DESCRIPTION
**Pull request by fxliang**

### CODE REVIEW

File: WeaselTSF/CandidateList.cpp [hunk 1]
```diff
@@ -423,7 +424,33 @@ void WeaselTSF::_SelectCandidateOnCurrentPage(size_t index)
...
+				_EndComposition(_pEditSessionContext, false);
+				// prediction situation, simuulate a Page_Up, to avoid crash
+				// fixme: are there better ways?
+				if (context->cinfo.candies.size()){
+					INPUT inputs[2];
+					inputs[0].type = INPUT_KEYBOARD;
+					inputs[0].ki = {VK_PRIOR, 0,0,0,0};
+					inputs[1].type = INPUT_KEYBOARD;
+					inputs[1].ki = {VK_PRIOR, 0,KEYEVENTF_KEYUP,0,0};
+					::SendInput(sizeof(inputs) / sizeof(INPUT), inputs, sizeof(INPUT));
```
Comment (lotem) [inline WeaselTSF/CandidateList.cpp:448]: I guess we can try to defer the `_UpdateComposition` call, or send an input key right here.

Sending an input key to the system essentially postpones the key event handling.

Comment (fxliang) [reply WeaselTSF/CandidateList.cpp:448]: if only  do `_UpdateComposition` here, crash is fixed, however the prediction now shown until next key press comes.

File: WeaselTSF/CandidateList.cpp [hunk 2]
```diff
@@ -423,7 +424,33 @@ void WeaselTSF::_SelectCandidateOnCurrentPage(size_t index)
 	m_client.SelectCandidateOnCurrentPage(index);
 	// fake a emptyp presskey to get data back and DoEditSession
 	m_client.ProcessKeyEvent(0);
-	_UpdateComposition(_pEditSessionContext);
```
Comment (lotem) [inline WeaselTSF/CandidateList.cpp:426]: I mean here, send a key event instead.

There is existing logic that does `GetResponseData`, `_UpdateLanguageBar` and insert commit text. If possible, make the minimal change by reusing existing code.

When sending a key event, try something without explicit behaviour, like `VK_PROCESSKEY`.

File: WeaselTSF/CandidateList.cpp (hunk 1 above)
Comment (lotem) [reply WeaselTSF/CandidateList.cpp:448]: How did the crash happen?

How is it fixed then?

Comment (fxliang) [reply WeaselTSF/CandidateList.cpp:448]: > How did the crash happen?
> 
> How is it fixed then?

maybe some object is referenced after released, after prediction is enabled and clicking select on non-prediction candidate

maybe some status different from space-selecting or auto-committing.

**Pull request merged**